lib/aurora_bootstrapper/notifier.rb

```diff
@@ -12,7 +12,59 @@ def export_date
     
     # ENV is string to string dictionary
     def export_date_override
-      DateTime.now.strftime("%Y-%m-%d") if ENV.key?('EXPORT_DATE_OVERRIDE')
+      datetime_subfolder_str = nil
+
+      if ENV.key?('EXPORT_DATE_OVERRIDE')
+        now = Date.today
+        
+        # expiration time is 30 days
+        for i in 1..30
+          datetime_subfolder_str = check_db_dump_done_for_one_day( datetime: (now-i))
+          unless datetime_subfolder_str.nil?
+            break
+          end
+        end
+
+        # the first run for the given db partition
+        if datetime_subfolder_str.nil?
+          datetime_subfolder_str = now.strftime("%Y-%m-%d")
+        end
+      end
+
+      datetime_subfolder_str
+    end
+
+    def check_db_dump_done_for_one_day( datetime: )
```

Comment:
This function:
* checks for files in a timestamped folder
* returns the passed in date, formatted

Can we separate the concerns and split into a couple functions? 

Also: what happens if there are files in the folder but no DONE file? does the function correctly reflect the `dump_done` part of its name.

Reply:
Separated.

Reply:
To answer your question, 

The code logic of export_date_override is updated as below if EXPORT_DATE_OVERRIDE is set:

Try to find the most recent datetime folder that contains DONE.txt

- if we find it, export_date is set to be that datetime plus one day
- if we do not find it, export_date is set to be today.

If the db dump job fails in the middle, then next run on the very next day will write the db dump files to the same folder. We will keep doing this till we successfully dump all the db files and write the DONE to the folder.

Reply:
The following would never occur:
2021-06-29 with data files and DONE file
2021-06-30 with data files but no DONE file
2021-07-01 with data files but no DONE file

Reply:
We could end up in one of the situations:

2021-06-29 with data files and DONE file
2021-06-30 with data files but no DONE file

or 

2021-06-29 with data files and DONE file
2021-06-30 with data files and DONE file